### ISSUE TITLE
Support TLS 1.3 for okhttp transport

### ISSUE THREAD

**Issue by voidzcy** [grpc/grpc-java#7431]
Our current okhttp version (2.7.4 Feb 2016) does not support TLS 1.3 https://github.com/grpc/grpc-java/blob/cb8b0679add2d389cfd15cd34e11c3577e3de2cb/okhttp/third_party/okhttp/main/java/io/grpc/okhttp/internal/ConnectionSpec.java#L60

Okhttp3 supports that. Since we don't have a short-term plan for upgrading to okhttp3, we may consider adding TLS 1.3 support to our existing okhttp transport.

**Comment by minopolous** [comment]
Yeah, same question we just fell over connection failures to a service which forces TLS 1.3 and found this ticket here.

**Comment by vinodhabib** [comment]
@MV Shiva Prasad I went through the details along with the current master code and found that TLS 1.3 changes are already in place as part of the PR merge (https://github.com/grpc/grpc-java/pull/8650).
I can see the TLS 1.3 support is added as part of ConnectionSpec along with CipherSuite (https://github.com/grpc/grpc-java/blob/d4c46a7f1fe67bad762139a679b3597dd8c1849d/okhttp/third_party/okhttp/main/java/io/grpc/okhttp/internal/ConnectionSpec.java#L61) which is internally using in OkHttpChannelBuilder as part of (https://github.com/grpc/grpc-java/blob/d4c46a7f1fe67bad762139a679b3597dd8c1849d/okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java#L111)

Also i tried by enable the commented code (https://github.com/grpc/grpc-java/blob/d4c46a7f1fe67bad762139a679b3597dd8c1849d/okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java#L129) and ran the complete build locally but i don't see any of the tests (currently not found the Http2OkhttpTest in master) are failing which is mentioned as part of issue (https://github.com/grpc/grpc-java/issues/8610) may be because default connection spec setup already using the TLS 1.3 internally as mentioned above (ConnectionSpec.java#L61)

Hope we are good to close this issue? please let me know how to proceed further on this and is there any pending action here?

**Comment by ejona86** [comment]
Yes, this was fixed by #8650.

Note that we still don't have TLS 1.3 enabled by default, but that is #7765. If we can uncomment those lines and it passes on Java 8, we'd like to do so, but I assume we'd still hit #7765.
